src/benchmarks/micro/libraries/System.IO.Hashing/CrcAppendTests.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Extensions;
+using MicroBenchmarks;
+
+namespace System.IO.Hashing.Tests
+{
+    [BenchmarkCategory(Categories.Libraries)]
+    public abstract class CrcAppendTests<T>
+        where T : NonCryptographicHashAlgorithm, new()
+    {
+        protected T Crc;
+        protected byte[] Buffer;
+
+        public abstract int BufferSize { get; set; }
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            Buffer = ValuesGenerator.Array<byte>(BufferSize);
+            Crc = new T();
+        }
+
+        [Benchmark]
+        public void Append()
+        {
+            Crc.Append(Buffer);
```

Comment:
Please excuse me if this is a dumb question (I've become an owner of `System.IO.Hashing` quite recently), but is my understanding correct that `Append` is not having any side-effects that could prolong the execution time of the next invocations and we don't need to call `Crc.Reset()` for every benchmark?